From a37c576c20b73a0c46d3da277b05d0ea770ea959 Mon Sep 17 00:00:00 2001
From: Juan Nunez-Iglesias <jni@fastmail.com>
Date: Sun, 21 May 2023 06:44:02 +1000
Subject: [PATCH] test: [Automatic] Constraints upgrades: `dask`, `hypothesis`,
 `torch` (#5835)

This PR is automatically created and updated by this napari GitHub
Action:

https://github.com/napari/napari/tree/main/.github/workflows/upgrade_test_contraints.yml
It ensures that dependencies are tested with the most recent version.

The updated packages are:

 * dask
 * fonttools
 * hypothesis
 * ipykernel
 * platformdirs
 * sympy
 * torch
 * tornado

Co-authored-by: alisterburt <alisterburt@users.noreply.github.com>
---
 resources/constraints/constraints_py3.10.txt  | 27 +++---
 resources/constraints/constraints_py3.8.txt   | 31 +++---
 resources/constraints/constraints_py3.9.txt   | 29 +++---
 .../constraints_py3.9_examples.txt            | 94 ++++++++-----------
 4 files changed, 86 insertions(+), 95 deletions(-)

diff --git a/resources/constraints/constraints_py3.10.txt b/resources/constraints/constraints_py3.10.txt
index 40b894ca..d0c2cd6b 100644
--- a/resources/constraints/constraints_py3.10.txt
+++ b/resources/constraints/constraints_py3.10.txt
@@ -48,7 +48,7 @@ comm==0.1.3
     # via ipykernel
 contourpy==1.0.7
     # via matplotlib
-coverage==7.2.7
+coverage==7.2.5
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
@@ -81,9 +81,9 @@ filelock==3.12.0
     #   virtualenv
 fonttools==4.39.3
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.5.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -95,7 +95,7 @@ hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.28.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -112,7 +112,7 @@ ipykernel==6.22.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.13.2
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -145,7 +145,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==16.0.3
     # via triton
 locket==1.0.0
     # via partd
@@ -266,7 +266,7 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.22
+pint==0.21
     # via napari (setup.cfg)
 platformdirs==3.7.0
     # via
@@ -381,7 +381,7 @@ pyzmq==25.0.2
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -392,11 +392,11 @@ qtpy==2.3.1
     #   napari-console
     #   qtconsole
     #   superqt
-requests==2.31.0
+requests==2.30.0
     # via
     #   pooch
     #   sphinx
-rich==13.4.2
+rich==13.3.5
     # via
     #   napari (setup.cfg)
     #   npe2
@@ -484,7 +484,7 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
 typing-extensions==4.5.0
     # via
@@ -495,11 +495,12 @@ typing-extensions==4.5.0
     #   pydantic
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.2
     # via requests
-virtualenv==20.23.1
+virtualenv==20.23.0
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
diff --git a/resources/constraints/constraints_py3.8.txt b/resources/constraints/constraints_py3.8.txt
index 7d3acd38..a12ae663 100644
--- a/resources/constraints/constraints_py3.8.txt
+++ b/resources/constraints/constraints_py3.8.txt
@@ -30,7 +30,7 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2022.12.7
+certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
@@ -48,11 +48,11 @@ comm==0.1.3
     # via ipykernel
 contourpy==1.0.7
     # via matplotlib
-coverage==7.2.7
+coverage==7.2.5
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
-dask==2023.5.0
+dask==2023.4.1
     # via napari (setup.cfg)
 debugpy==1.6.7
     # via ipykernel
@@ -81,9 +81,9 @@ filelock==3.12.0
     #   virtualenv
 fonttools==4.39.3
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.5.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -95,7 +95,7 @@ hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.28.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -119,7 +119,7 @@ ipykernel==6.22.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.12.0
+ipython==8.12.2
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -152,7 +152,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==16.0.3
     # via triton
 locket==1.0.0
     # via partd
@@ -273,7 +273,7 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.21.1
+pint==0.21
     # via napari (setup.cfg)
 pkgutil-resolve-name==1.3.10
     # via jsonschema
@@ -392,7 +392,7 @@ pyzmq==25.0.2
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -403,11 +403,11 @@ qtpy==2.3.1
     #   napari-console
     #   qtconsole
     #   superqt
-requests==2.31.0
+requests==2.30.0
     # via
     #   pooch
     #   sphinx
-rich==13.4.2
+rich==13.3.5
     # via
     #   napari (setup.cfg)
     #   npe2
@@ -495,7 +495,7 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
 typing-extensions==4.5.0
     # via
@@ -508,11 +508,12 @@ typing-extensions==4.5.0
     #   rich
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.2
     # via requests
-virtualenv==20.23.1
+virtualenv==20.23.0
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
diff --git a/resources/constraints/constraints_py3.9.txt b/resources/constraints/constraints_py3.9.txt
index e3aac73f..e32398cc 100644
--- a/resources/constraints/constraints_py3.9.txt
+++ b/resources/constraints/constraints_py3.9.txt
@@ -30,7 +30,7 @@ build==0.10.0
     # via npe2
 cachey==0.2.1
     # via napari (setup.cfg)
-certifi==2022.12.7
+certifi==2023.5.7
     # via
     #   napari (setup.cfg)
     #   requests
@@ -48,7 +48,7 @@ comm==0.1.3
     # via ipykernel
 contourpy==1.0.7
     # via matplotlib
-coverage==7.2.7
+coverage==7.2.5
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
@@ -81,9 +81,9 @@ filelock==3.12.0
     #   virtualenv
 fonttools==4.39.3
     # via matplotlib
-freetype-py==2.3.0
+freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.5.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -95,7 +95,7 @@ hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.28.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
@@ -117,7 +117,7 @@ ipykernel==6.22.0
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.13.2
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -150,7 +150,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==16.0.3
     # via triton
 locket==1.0.0
     # via partd
@@ -271,7 +271,7 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.22
+pint==0.21
     # via napari (setup.cfg)
 platformdirs==3.7.0
     # via
@@ -386,7 +386,7 @@ pyzmq==25.0.2
     #   ipykernel
     #   jupyter-client
     #   qtconsole
-qtconsole==5.4.1
+qtconsole==5.4.3
     # via
     #   napari (setup.cfg)
     #   napari-console
@@ -397,11 +397,11 @@ qtpy==2.3.1
     #   napari-console
     #   qtconsole
     #   superqt
-requests==2.31.0
+requests==2.30.0
     # via
     #   pooch
     #   sphinx
-rich==13.4.2
+rich==13.3.5
     # via
     #   napari (setup.cfg)
     #   npe2
@@ -489,7 +489,7 @@ traitlets==5.9.0
     #   qtconsole
 triton==2.0.0
     # via torch
-typer==0.7.0
+typer==0.9.0
     # via npe2
 typing-extensions==4.5.0
     # via
@@ -501,11 +501,12 @@ typing-extensions==4.5.0
     #   pydantic
     #   superqt
     #   torch
+    #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.2
     # via requests
-virtualenv==20.23.1
+virtualenv==20.23.0
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
diff --git a/resources/constraints/constraints_py3.9_examples.txt b/resources/constraints/constraints_py3.9_examples.txt
index 0d2312a9..1349dee6 100644
--- a/resources/constraints/constraints_py3.9_examples.txt
+++ b/resources/constraints/constraints_py3.9_examples.txt
@@ -42,13 +42,13 @@ click==8.1.3
     #   typer
 cloudpickle==2.2.1
     # via dask
-cmake==3.26.4
+cmake==3.26.3
     # via triton
 comm==0.1.3
     # via ipykernel
-contourpy==1.1.0
+contourpy==1.0.7
     # via matplotlib
-coverage==7.2.7
+coverage==7.2.5
     # via pytest-cov
 cycler==0.11.0
     # via matplotlib
@@ -74,7 +74,7 @@ executing==1.2.0
     # via stack-data
 fasteners==0.18
     # via zarr
-filelock==3.12.2
+filelock==3.12.0
     # via
     #   torch
     #   triton
@@ -83,7 +83,7 @@ fonttools==4.40.0
     # via matplotlib
 freetype-py==2.4.0
     # via vispy
-fsspec==2023.6.0
+fsspec==2023.5.0
     # via
     #   dask
     #   napari (setup.cfg)
@@ -95,14 +95,14 @@ hypothesis==6.79.1
     # via napari (setup.cfg)
 idna==3.4
     # via requests
-imageio==2.31.1
+imageio==2.28.1
     # via
     #   napari (setup.cfg)
     #   napari-svg
     #   scikit-image
 imagesize==1.4.1
     # via sphinx
-importlib-metadata==6.7.0
+importlib-metadata==6.6.0
     # via
     #   dask
     #   jupyter-client
@@ -117,7 +117,7 @@ ipykernel==6.23.2
     # via
     #   napari-console
     #   qtconsole
-ipython==8.14.0
+ipython==8.13.2
     # via
     #   ipykernel
     #   napari (setup.cfg)
@@ -141,7 +141,7 @@ jupyter-client==8.2.0
     # via
     #   ipykernel
     #   qtconsole
-jupyter-core==5.3.1
+jupyter-core==5.3.0
     # via
     #   ipykernel
     #   jupyter-client
@@ -154,7 +154,7 @@ lazy-loader==0.2
     # via
     #   napari (setup.cfg)
     #   scikit-image
-lit==16.0.6
+lit==16.0.3
     # via triton
 locket==1.0.0
     # via partd
@@ -164,9 +164,9 @@ lxml==4.9.2
     #   nilearn
 magicgui==0.7.2
     # via napari (setup.cfg)
-markdown-it-py==3.0.0
+markdown-it-py==2.2.0
     # via rich
-markupsafe==2.1.3
+markupsafe==2.1.2
     # via jinja2
 matplotlib==3.7.1
     # via napari (setup.cfg)
@@ -187,9 +187,7 @@ napari-plugin-engine==0.2.0
     #   napari (setup.cfg)
     #   napari-console
     #   napari-svg
-napari-plugin-manager==0.1.0a2
-    # via napari (setup.cfg)
-napari-svg==0.1.10
+napari-svg==0.1.6
     # via napari (setup.cfg)
 nest-asyncio==1.5.6
     # via ipykernel
@@ -202,9 +200,7 @@ nibabel==5.1.0
 nilearn==0.10.1
     # via -r resources/constraints/version_denylist_examples.txt
 npe2==0.7.0
-    # via
-    #   napari (setup.cfg)
-    #   napari-plugin-manager
+    # via napari (setup.cfg)
 numcodecs==0.11.0
     # via zarr
 numpy==1.23.5
@@ -274,7 +270,7 @@ packaging==23.1
     #   superqt
     #   vispy
     #   xarray
-pandas==2.0.2 ; python_version >= "3.9"
+pandas==2.0.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   nilearn
@@ -293,14 +289,14 @@ pillow==9.5.0
     #   matplotlib
     #   napari (setup.cfg)
     #   scikit-image
-pint==0.22
+pint==0.21
     # via napari (setup.cfg)
 platformdirs==3.7.0
     # via
     #   jupyter-core
     #   pooch
     #   virtualenv
-pluggy==1.2.0
+pluggy==1.0.0
     # via pytest
 pooch==1.7.0
     # via
@@ -312,7 +308,7 @@ psutil==5.9.5
     # via
     #   ipykernel
     #   napari (setup.cfg)
-psygnal==0.9.1
+psygnal==0.9.0
     # via
     #   app-model
     #   magicgui
@@ -322,7 +318,7 @@ ptyprocess==0.7.0
     # via pexpect
 pure-eval==0.2.2
     # via stack-data
-pydantic==1.10.9
+pydantic==1.10.7
     # via
     #   app-model
     #   napari (setup.cfg)
@@ -336,10 +332,8 @@ pygments==2.15.1
     #   sphinx
     #   superqt
 pyopengl==3.1.6
-    # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
-pyparsing==3.1.0
+    # via napari (setup.cfg)
+pyparsing==3.0.9
     # via matplotlib
 pyproject-hooks==1.0.0
     # via build
@@ -349,9 +343,9 @@ pyqt5-qt5==5.15.2
     # via pyqt5
 pyqt5-sip==12.12.1
     # via pyqt5
-pyqt6==6.5.1
+pyqt6==6.5.0
     # via napari (setup.cfg)
-pyqt6-qt6==6.5.1
+pyqt6-qt6==6.5.0
     # via pyqt6
 pyqt6-sip==13.5.1
     # via pyqt6
@@ -369,7 +363,7 @@ pyside6-essentials==6.3.1
     # via
     #   pyside6
     #   pyside6-addons
-pytest==7.3.2
+pytest==7.3.1
     # via
     #   napari (setup.cfg)
     #   pytest-cov
@@ -377,13 +371,13 @@ pytest==7.3.2
     #   pytest-metadata
     #   pytest-pretty
     #   pytest-qt
-pytest-cov==4.1.0
+pytest-cov==4.0.0
     # via
     #   -r resources/constraints/version_denylist.txt
     #   napari (setup.cfg)
 pytest-json-report==1.5.0
     # via -r resources/constraints/version_denylist.txt
-pytest-metadata==3.0.0
+pytest-metadata==2.0.4
     # via pytest-json-report
 pytest-pretty==1.2.0
     # via napari (setup.cfg)
@@ -405,7 +399,7 @@ pyyaml==6.0
     #   dask
     #   napari (setup.cfg)
     #   npe2
-pyzmq==25.1.0
+pyzmq==25.0.2
     # via
     #   ipykernel
     #   jupyter-client
@@ -419,24 +413,23 @@ qtpy==2.3.1
     #   magicgui
     #   napari (setup.cfg)
     #   napari-console
-    #   napari-plugin-manager
     #   qtconsole
     #   superqt
-requests==2.31.0
+requests==2.30.0
     # via
     #   nilearn
     #   pooch
     #   sphinx
-rich==13.4.2
+rich==13.3.5
     # via
     #   napari (setup.cfg)
     #   npe2
     #   pytest-pretty
-scikit-image==0.21.0
+scikit-image==0.20.0
     # via napari (setup.cfg)
 scikit-learn==1.2.2
     # via nilearn
-scipy==1.10.1 ; python_version >= "3.9"
+scipy==1.9.1 ; python_version >= "3.9"
     # via
     #   napari (setup.cfg)
     #   nilearn
@@ -482,10 +475,8 @@ superqt==0.4.1
     #   napari-plugin-manager
 sympy==1.12
     # via torch
-tensorstore==0.1.37
-    # via
-    #   -r resources/constraints/version_denylist.txt
-    #   napari (setup.cfg)
+tensorstore==0.1.36
+    # via napari (setup.cfg)
 threadpoolctl==3.1.0
     # via scikit-learn
 tifffile==2023.4.12
@@ -503,11 +494,11 @@ toolz==0.12.0
     #   dask
     #   napari (setup.cfg)
     #   partd
-torch==2.0.1
+torch==2.0.0
     # via
     #   napari (setup.cfg)
     #   triton
-tornado==6.3.2
+tornado==6.3.1
     # via
     #   ipykernel
     #   jupyter-client
@@ -526,13 +517,12 @@ triton==2.0.0
     # via torch
 typer==0.9.0
     # via npe2
-typing-extensions==4.6.3
+typing-extensions==4.5.0
     # via
     #   app-model
     #   ipython
     #   magicgui
     #   napari (setup.cfg)
-    #   pint
     #   psygnal
     #   pydantic
     #   superqt
@@ -540,9 +530,9 @@ typing-extensions==4.6.3
     #   typer
 tzdata==2023.3
     # via pandas
-urllib3==2.0.3
+urllib3==2.0.2
     # via requests
-virtualenv==20.23.1
+virtualenv==20.23.0
     # via napari (setup.cfg)
 vispy==0.12.2
     # via
@@ -560,9 +550,9 @@ wheel==0.40.0
     #   nvidia-nvtx-cu11
 wrapt==1.15.0
     # via napari (setup.cfg)
-xarray==2023.5.0
+xarray==2023.4.2
     # via napari (setup.cfg)
-zarr==2.15.0
+zarr==2.14.2
     # via napari (setup.cfg)
 zipp==3.15.0
     # via
@@ -570,9 +560,7 @@ zipp==3.15.0
     #   importlib-resources
 
 # The following packages are considered to be unsafe in a requirements file:
-pip==23.1.2
-    # via napari-plugin-manager
-setuptools==68.0.0
+setuptools==67.7.2
     # via
     #   nvidia-cublas-cu11
     #   nvidia-cuda-cupti-cu11
-- 
2.40.0.windows.1
